src/main/java/core/basesyntax/FlipBit.java

```diff
@@ -9,6 +9,20 @@
      * Договоримся, что биты нумеруются от младшего (индекс 1) к старшему (индекс 32).</p>
      */
     public int flipBit(int value, int bitIndex) {
-        return 0;
+        StringBuilder stringBuilderBinaryValue = new StringBuilder(Integer.toBinaryString(value));
+        stringBuilderBinaryValue.reverse();
+        while (stringBuilderBinaryValue.length() < 32) {
+            stringBuilderBinaryValue.append('0');
+        }
+        stringBuilderBinaryValue.reverse();
+        int indexChangeBite = stringBuilderBinaryValue.length() - bitIndex;
+        if (stringBuilderBinaryValue.charAt(indexChangeBite) == '1') {
+            stringBuilderBinaryValue.replace(indexChangeBite,
+                    indexChangeBite + 1, String.valueOf('0'));
+        } else {
+            stringBuilderBinaryValue.replace(indexChangeBite,
+                    indexChangeBite + 1, String.valueOf('1'));
+        }
+        return Integer.parseInt(stringBuilderBinaryValue.toString(), 2);
```

Comment:
Good job! now that you understand how it works you can simplify this solution using XOR and left shift